File: publisher.js
var zmq = require('zmq');
var sock = zmq.socket('pub');
var port = 5556;
var count = 0;

function sendMessage(topic, data) {
	var message = [topic, JSON.stringify(data)].join(' ');
	console.log('Sending message: %s', message);
	sock.send(message);
};

sock.bind('tcp://127.0.0.1:' + port, function(err) {
	if (err) {
		console.error("Could not bind to port %s", port);

	} else {
		console.log("Bound to port %s", port);
		setInterval(function(){
			sendMessage('seconds.ping', {
				time: new Date().toISOString()
			});
			if (count == 0) {
				sendMessage('minutes.ping', {
					time: new Date().toISOString()
				});
			}
			count = (count + 1) % 60;
		}, 1000);
	}
});

File: publisher.py
import zmq
import time
import json

port = 5556
count = 0

context = zmq.Context()
socket = context.socket(zmq.PUB)
socket.bind("tcp://127.0.0.1:%s" % port)

def send_json_message(socket, topic, message):
	print("Sending message: %s %s" % (topic, json.dumps(message)))
	socket.send_string("%s %s" % (topic, json.dumps(message)))

while True:
	send_json_message(socket, "seconds.ping", {
		'time': time.strftime('%Y-%m-%dT%H:%M:%SZ', time.gmtime())
	})
	
	if count == 0:
		send_json_message(socket, "minutes.ping", {
			'time': time.strftime('%Y-%m-%dT%H:%M:%SZ', time.gmtime())
		})

	count = (count + 1) % 60
	time.sleep(1)
